addons/web/static/src/js/view_list.js

```diff
@@ -1872,7 +1872,7 @@ var Record = instance.web.Class.extend(/** @lends Record# */{
             if (typeof val !== 'object') {
                 output[k] = val;
             } else if (val instanceof Array) {
-                output[k] = val[0];
+                output[k] = val.length > 0 ? val[0] : null;
```

Comment:
Gotcha, thanks @pedrobaeza 

May I suggest a comment of such? I feel like someone working in this section will always have this question & could end us with regressions.